### DISCUSSION TITLE
GitLab open sources secret detection in video content under MIT 🎥 🔒

### DISCUSSION THREAD

**Discussion by JamieSlome** [finos/git-proxy#525] [Ideas]
GitLab's recently open sourced [secret detection in video content with OCR](https://about.gitlab.com/blog/2024/02/29/how-to-detecting-secrets-in-video-content/):

<a href="https://about.gitlab.com/blog/2024/02/29/how-to-detecting-secrets-in-video-content/"><img width="835" alt="Screenshot 2024-04-12 at 11 12 46" src="https://github.com/finos/git-proxy/assets/55323451/2ef9bb0a-d6b4-4a0b-9712-f3dc4badf57a"></a>

This could be an interesting plugin. Thoughts? 💬

**Comment by maoo** [comment]
Very interesting indeed!

I think it really depends here on what delivers the most value.

My personal opinion - I believe that the 2.x version of GitProxy will open an infinite amount of opportunities and we could plug scanning logic about pretty much anything, so it's fair to ask ourselves:
- How/where do we collect these opportunities?
- Shall these plugins be developed by the GitProxy core team?
- How do we prioritize these activities?

To start, I'd suggest to start simple and allocate a GitHub issue label (we could call it `plugin proposal` or something like that; proposals should IMO commented by consumers (ie Citi, RBC, NatWest) to understand if it's of interest for anyone using GitProxy today.

As a second step, I'd try to help, support, incentivate the community to pick up these tasks, and events like Hackathons or Tech Sprints could be a good addition.

My 2 cents of course, eager to hear your thoughts!
